code/modules/vehicles/wheelchair.dm

```diff
@@ -67,3 +80,68 @@
 /obj/vehicle/ridden/wheelchair/proc/make_ridable()
 	AddElement(/datum/element/ridable, /datum/component/riding/vehicle/wheelchair)
 
+// Battlechair - A wheelchair with a mounted minigun
+/obj/vehicle/ridden/wheelchair/weaponized
+	name = "\improper Battlechair"
+	desc = "A sturdy wheelchair fitted with a minigun. Your legs may have failed you, but your weapon won't."
+	max_integrity = 400
+
+	///Reference to the mounted weapon
+	var/obj/item/weapon/gun/minigun/wheelchair/weapon
```

Comment:
I still think all this vehicle weapon mount stuff should be generic instead of only on this type. It has application for other vehicles like bike side cars or whatever.